opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/otlp_encoder.rs

```diff
@@ -245,6 +367,95 @@ impl OtlpEncoder {
         (field_defs, schema_id)
     }
 
+    /// Determine span fields and calculate schema ID in a single pass for optimal performance
+    fn determine_span_fields_and_schema_id(span: &Span, event_name: &str) -> (Vec<FieldDef>, u64) {
+        use std::collections::hash_map::DefaultHasher;
+        use std::hash::{Hash, Hasher};
+
+        // Pre-allocate with estimated capacity to avoid reallocations
+        let estimated_capacity = 12 + span.attributes.len(); // 7 always + 5 max conditional + attributes
```

Comment:
This calculation seems wrong. It looks like you have missed the three Part A conditional fields.